contrib/kind.sh

```diff
@@ -205,307 +203,339 @@ print_params()
+  # Docker disables IPv6 globally inside containers except in the eth0 interface.
+  # Kind enables IPv6 globally the containers ONLY for dual-stack and IPv6 deployments.
+  # Ovnkube-node tries to move all global addresses from the gateway interface to the
+  # bridge interface it creates. This breaks on KIND with IPv4 only deployments, because the new
+  # internal bridge has IPv6 disable and can't move the IPv6 from the eth0 interface.
+  # We can enable IPv6 always in the container, since the docker setup with IPv4 only
+  # is not very common.
+  KIND_NODES=$(kind get nodes --name ${KIND_CLUSTER_NAME})
+  for n in $KIND_NODES; do
+    docker exec $n sysctl net.ipv6.conf.all.disable_ipv6=0
+    docker exec $n sysctl net.ipv6.conf.all.forwarding=1
+  done
+}
 
-ovn_apiServerAddress=${API_IP} \
-  ovn_ip_family=${IP_FAMILY} \
-  ovn_ha=${OVN_HA} \
-  net_cidr=${NET_CIDR} \
-  svc_cidr=${SVC_CIDR} \
-  ovn_num_master=${KIND_NUM_MASTER} \
-  ovn_num_worker=${KIND_NUM_WORKER} \
-  cluster_log_level=${KIND_CLUSTER_LOGLEVEL:-4} \
-  j2 ${KIND_CONFIG} -o ${KIND_CONFIG_LCL}
+coredns_patch_for_github_ci() {
+  if [ "${GITHUB_ACTIONS:-false}" == "true" ]; then
+    # Patch CoreDNS to work in Github CI
+    # 1. Github CI doesn´t offer IPv6 connectivity, so CoreDNS should be configured
+    # to work in an offline environment:
+    # https://github.com/coredns/coredns/issues/2494#issuecomment-457215452
+    # 2. Github CI adds following domains to resolv.conf search field:
+    # .net.
+    # CoreDNS should handle those domains and answer with NXDOMAIN instead of SERVFAIL
+    # otherwise pods stops trying to resolve the domain.
+    # Get the current config
+    original_coredns=$(kubectl get -oyaml -n=kube-system configmap/coredns)
+    echo "Original CoreDNS config:"
+    echo "${original_coredns}"
+    # Patch it
+    fixed_coredns=$(
+      printf '%s' "${original_coredns}" | sed \
+        -e 's/^.*kubernetes cluster\.local/& net/' \
+        -e '/^.*upstream$/d' \
+        -e '/^.*fallthrough.*$/d' \
+        -e '/^.*forward . \/etc\/resolv.conf$/d' \
+        -e '/^.*loop$/d' \
+    )
+    echo "Patched CoreDNS config:"
+    echo "${fixed_coredns}"
+    printf '%s' "${fixed_coredns}" | kubectl apply -f -
+  fi
+}
 
-# Create KIND cluster. For additional debug, add '--verbosity <int>': 0 None .. 3 Debug
-export KUBECONFIG=${HOME}/admin.conf
-if kind get clusters | grep ovn; then
-  delete
-fi
-kind create cluster --name ${KIND_CLUSTER_NAME} --kubeconfig ${KUBECONFIG} --image kindest/node:${K8S_VERSION} --config=${KIND_CONFIG_LCL}
-cat ${KUBECONFIG}
+build_ovn_image() {
+  if [ "$OVN_IMAGE" == local ]; then
+    # Build ovn docker image
+    pushd ../go-controller
+    make
+    popd
+
+    # Build ovn kube image
+    pushd ../dist/images
+    # Find all built executables, but ignore the 'windows' directory if it exists
+    BINS=$(find ../../go-controller/_output/go/bin/ -maxdepth 1 -type f | xargs)
+    sudo cp -f ${BINS} .
```

Comment:
I was going to bring this up separately, but since you're working on kind.sh, why does this need to be sudo?  This is a pain because every time I run this I have to enter my sudo password half-way through.  It seems like this line is just a copy of some binaries from one subdirectory in ovn-kubernetes/ to another.  If I'm running this as a normal user, then I own both the source and dest directories.

Reply:
I agree. will remove it